Objects/dictobject.c

```diff
@@ -3005,21 +3005,28 @@ dictview_repr(dictviewobject *dv)
 {
     PyObject *seq;
     PyObject *seq_str;
-    PyObject *result;
+    PyObject *result = NULL;
+    Py_ssize_t rc;
 
+    rc = Py_ReprEnter((PyObject *)dv);
+    if (rc != 0) {
+        return rc > 0 ? PyString_FromString("...") : NULL;
+    }
     seq = PySequence_List((PyObject *)dv);
-    if (seq == NULL)
-        return NULL;
-
+    if (seq == NULL) {
+        goto Done;
+    }
     seq_str = PyObject_Repr(seq);
     if (seq_str == NULL) {
-        Py_DECREF(seq);
-        return NULL;
+        goto Done_seq;
```

Comment:
`Py_DECREF(seq)` could be called just after `PyObject_Repr()`. This will simplify the code.

Reply:
Thanks for suggestion; that is indeed cleaner.  I've just force-pushed a fresh commit including that change.